src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCITriggerService.java

```diff
@@ -228,7 +238,15 @@ private RepositoryInfo getRepositoryInfo(ProgrammingExerciseParticipation partic
         String[] auxiliaryRepositoryUris = auxiliaryRepositories.stream().map(AuxiliaryRepository::getRepositoryUri).toArray(String[]::new);
         String[] auxiliaryRepositoryCheckoutDirectories1 = auxiliaryRepositories.stream().map(AuxiliaryRepository::getCheckoutDirectory).toArray(String[]::new);
 
-        if (programmingExercise.getCheckoutSolutionRepository()) {
+        ProgrammingExerciseBuildConfig buildConfig;
+        if (Hibernate.isInitialized(programmingExercise.getBuildConfig())) {
```

Comment:
Why no null check here?

Reply:
(I looked it up and Hibernate.isInitialized returns `true` for `null` values. We may indeed want to check for that here)